src/Education.UnitTests/Commands/ShowRuleHelpFromErrorListTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using FluentAssertions;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Education.Commands;
+using SonarLint.VisualStudio.Infrastructure.VS;
+using SonarLint.VisualStudio.Integration.UnitTests;
+
+namespace SonarLint.VisualStudio.Education.UnitTests.Commands
+{
```

Comment:
I wanted to add a test ->

```  
[TestMethod]
 public void MefCtor_CheckIsExported()
  {
            MefTestHelpers.CheckTypeCanBeImported<ShowHelpFromErrorList, IShowHelpFromErrorList>(
                MefTestHelpers.CreateExport<SVsServiceProvider>(),
                MefTestHelpers.CreateExport<IEducation>(),
                MefTestHelpers.CreateExport<IErrorListHelper>());
}
```

However, this calls the ctor, which calls "InitializeInterception()", which uses the threadhandler. This then throws an exception. Please let me know how to deal with this :)

Reply:
Ah. One option would be to MEF import `IThreadHandling` , and in `CreateExport` pass in the no-op thread handling implementation.

You would need to add a new overload to `CreateExport` since the current one is hard-coded to use a dummy value:

```
        [DebuggerStepThrough]
        public static Export CreateExport<T>() where T: class
        {
            return CreateExport<T>(Mock.Of<T>());
        }
```

(there did use to be an overload that accepted the object to export, but I removed it because it wasn't being used).
However, you would still need to provide a minimal implementation for the service provider and `IVsRegisterPriorityCommandTarget`, otherwise that code will fail too.

An alternative would be to leave the test as it is, but catch the exception and check it is null-ref exception. It's a bit ugly, but it would be enough to tell us what we want to know i.e. that the class was created via MEF (if the class can't be created via MEF we'd get a `CompositionException` instead).